src/App.js

```diff
@@ -0,0 +1,67 @@
+import React, { useEffect } from "react";
+import {Routes, Route} from 'react-router-dom';
+import LoginPage from "./pages/LoginPage";
+import RegisterPage from "./pages/RegisterPage";
+import HomePage from "./pages/Homepage";
+import AddPage from "./pages/AddPage";
+import DetailPage from "./pages/DetailPage";
+import Navigation from "./components/Navigation";
+import Loading from "./components/Loading";
+import NotFoundPage from './pages/NotFoundPage';
+import { useSelector,useDispatch } from "react-redux";
+import { asyncPreloadProcess } from './states/isPreload/action';
+import { asyncUnsetAuthUser } from './states/authUser/action';
+function App() {
+  const {
+    authUser = null,
+    isPreload = false
+  } = useSelector((states) => states)
+
+  const dispatch = useDispatch()
+
+  useEffect(() => {
+    dispatch(asyncPreloadProcess());
+  }, [dispatch]);
+
+  const onSignOut = () => {   
+    dispatch(asyncUnsetAuthUser());
+  };
```

Comment:
```suggestion
  const onSignOut = () => dispatch(asyncUnsetAuthUser());
```
Kamu bisa menuliskan arrow function seperti ini jika hanya terdapat satu baris kode didalam function tersebut